sktime/forecasting/hf_patchtst_forecaster.py

```diff
@@ -0,0 +1,507 @@
+            or the [`~PreTrainedModel.save_pretrained`] method
+            - A path or url to a saved configuration JSON *file*, e.g.,
+                `./my_model_directory/configuration.json`.
+            forecast_columns,
+    mode : str, optional, possible values = ["untrained", "finetune", "zeroshot"]
+           default = "untrained"
+        String to set the mode of the model. If set to 'untrained', it will
+        re-initialize an untrained model with the specified config or
+        estimator aruguments. If set to "finetune" will use the `model_path`
+        argument and the passed in `y` in fit to fine-tune the model. If
+        set to "zeroshot", it will load the model in zero-shot forecasting
+        model with the argument `model_path` and ignore any passed `y`.
+        Note that both "finetune" and "zeroshot" mode requires a mandatory
+        passed in `model_path`.
+    patch_length : int, optional, default = 2
+        Length of each patch that will segment every univariate series.
+    context_length : int, optional, default = 4
+        Number of previous time steps used to forecast.
+    patch_stride : int, optional, default = 2
+        Length of the non-overlapping region between patches. If patch_stride
+        is less than patch_length, then there will be overlapping patches. If
+        patch_stride = patch_length, then there will be no overlapping patches.
+    random_mask_ratio : float, optional, default = 0.4
+        Masking ratio applied to mask the input data during random pretraining.
+    d_model : int, optional, default = 128
+        Dimension of the weight matrices in the transformer layers.
+    num_attention_heads : int, optional, default = 16
+        Number of attention heads for each attention layer in the transformer block
+    ffn_dim : int, optional, default = 256
+        Dimensionality of the feed forward layer in the transformer encoder.
+    head_dropout : float, optional, default = 0.2
+        Dropout probability for a head.
+    batch_size : int, optional, default = 64
+        Size of every batch during training. Reduce if you have reduced gpu power.
+    learning_rate : float, optional, default = 1e-4
+        Leaning rate that is used during training.
+    epochs : int, optional, default = 10
+        Number of epochs to use during training.
+    validation_split : float, optional, default = 0.2
+        Fraction of the data to use for validation.
+    config : dict, optional, default = {}
+        A config dict specifying parameters to initialize an untrained
+        PatchTST model. Missing parameters in the config will be automatically
+        replaced by their default values. See the PatchTSTConfig config on
+        huggingface for more details.
+    training_args : dict, optional, default = None
+        Training arguments to use for the model. If this is passed,
+        the remaining applicable training arguments will be ignored
```

Comment:
`patch_length`, `context_length`, `patch_stride`, .... why put these parameters explicitly when they are already part of `config`?

similarly with `epochs`, `learning_rate`, `batch_size`. they are a part of `training_args`

Reply:
The goal of these extra explicit parameters is to ease user configurability, instead of having to force the user to create their own config dictionary, they can either

1) Initialize the model explicitly with default values, passing nothing
2) passing simple model parameters or training parameters, without having to do any extra work

For more complex model and training scenarios, the user has the option to pass in their own configured dictionaries for the model config and the training args config

Reply:
I still think, params from config should not be explicit. This creates confusion and repetition. Similar case was with `HFTransformerForecaster` and users can pass custom config and training config implicitly.

Reply:
I've made it clear in the docstring that if `config` is passed, it will override any parameters specified in the `init` args, to avoid confusion or repetition. If you feel like that still is not a good idea, I'll remove them

Reply:
I agree with @geetu040. I suppose it is better to only have the config if possible and mention to check out the transformers library for setting the parameters.

Reply:
Thanks for your input Benedikt and Armaghan, I'll change it shortly.